tests/test_blocks.py

```diff
@@ -162,3 +167,14 @@ def test_block_windows_filtered_none(path_rgb_byte_tif):
         itr = ((ij, win) for ij, win in src.block_windows() if filter_func(win))
         with pytest.raises(StopIteration):
             next(itr)
+
+@pytest.mark.skipif(
+    parse(rasterio.__gdal_version__) < parse('2.0.0'),
+    reason="TIFF block size access requires GDAL 2.0")
+def test_block_tiff(path_rgb_byte_tif):
+    """Without compression a TIFF's blocks are all the same size"""
+    with rasterio.open(path_rgb_byte_tif) as src:
+        block_windows = list(src.block_windows())
+        sizes = [src.block(1, i, j).size for (i, j), w in block_windows]
+        assert sizes.count(2373) == 1
+        assert sizes.count(7119) == len(block_windows) - 1
```

Comment:
Can we include a test for a non-TIFF raster? That should be able to run on GDAL 1.x, correct?